src/checks.cc

```diff
@@ -436,7 +462,25 @@ LinterResult CheckCountStar(absl::string_view sql,
 LinterResult CheckKeywordNamedIdentifier(absl::string_view sql,
                                          const LinterOptions &option) {
   LinterResult result;
+  std::vector<ParseToken> keywords = GetKeywords(sql);
```

Comment:
If you choose my suggestion to extract file parsing into separate function, the second step would be to extract all keywords and identifiers also only once, before all checks :)

Reply:
I think "RunChecks" is a seperate function for adding sth. before all checks. So, I can add one line to it to save keywords/Tokens etc. But, I don't think it would be effective. Most checks needs to traverse AST tree to apply specific functions and just traversing is not costly at all compared to whole parser. As for tokenizer, There is literally 2 checks that using it.

So, I think the can be added to options to optimize by creating these only once, but it doesn't change much (20 checks having ~O(4N), we will have ~O(N/10) identifiers that will be created twice, I could say roughly <1%). I am leaving it to future.

Reply:
sounds good